### ISSUE TITLE
Private key is exposed when specifying a key for token functionality via TokenCreate precompile

### ISSUE THREAD

**Issue by ed-marquez** [hashgraph/hedera-smart-contracts#66]
### Description

The following bit of code:
```
 let fcnParams = new ContractFunctionParameters()
        .addBytes(operatorPvKey.toBytes())
```
- specifies the supply key (and some other properties) for an HTS token that is created via TokenCreate precompile
- passes a private key instead of the public one (this could be done by choice or accidentally)
- **ISSUE:** exposes that private key
- Results in: https://testnet.mirrornode.hedera.com/api/v1/tokens/0.0.48224520 , where the supply key shown is `ed8f85706372da5a1f4d3c8b409187e3278dd2087a5c86cefc96a55823d0987e `- that's `operatorPvKey.toStringRaw()`
From that, running `PrivatePvKey.fromStringED25519("ed8f85706372da5a1f4d3c8b409187e3278dd2087a5c86cefc96a55823d0987e")` shows the key now in `302...`  Hedera format... **Effectively this exposes the private key the user specified.**❌

Specifying the supply key by passing the public key works as expected, exposing only the public key:
```
let fcnParams = new ContractFunctionParameters()
        .addBytes(operatorPvKey.publicKey.toBytes())
```
Results in this token: https://testnet.mirrornode.hedera.com/api/v1/tokens/0.0.48224494 , where the supply key shown is `d918da30b825d0cdcc1fff01ae5cc9716c2e1d2d8b9e3ab8c7a50755de6a5c3b `- basically, `operatorPvKey.publicKey.toStringRaw()` ✅

### Steps to reproduce

1. Go to: https://gitpod.io/#https://github.com/ed-marquez/hedera-example-contract-tokenCreate-precompile
2. add a .env file and appropriate credentials
3. See line 70 in `index.js` - change that line to pass the private key instead of the public key
4. Run `index.js` and inspect the token properties via mirror node - https://testnet.mirrornode.hedera.com/api/v1/tokens/0.0.xxxxx
5. See private key being exposed

### Additional context

Users accustomed to creating tokens via the SDK who transition to creating tokens via TokenCreate precompile may inadvertently specify a private key for critical functionality of their  token. In that case,

- Tokens created via the SDK would show the public key that corresponds to the private key controlling certain functionality (as it should be) ❌

- Tokens created via precompile would expose that private key (this should never be the case - only the corresponding public key should be shown)✅

In other words, if a user passes a key as a contract function parameter to control certain token functionality, whether operatorPvKey or operatorPvKey.publicKey - **the private key should never be exposed to the public.**

### Hedera network

mainnet, testnet, previewnet

### Version

2.18

### Operating system

Windows

**Comment by gregscullard** [comment]
@ed-marquez "In other words, if a user passes a key as a contract function parameter to control certain token functionality, whether operatorPvKey or operatorPvKey.publicKey - the private key should never be exposed to the public."

If the private key was passed to the contract, it will exist inside the corresponding transaction and is therefore already exposed to the public. 
Wondering if the operation should fail unless a public key is specified...

**Comment by Nana-EC** [comment]
Thanks @ed-marquez, unfortunately I do believe this falls under developer security hygiene.

Hedera APIs will never ask for private keys as parameters.
Developers need only ever use their private keys for signing transactions locally in the SDK or with their web3 wallet.
This same issue is present on Ethereum for example if you provided private key data as a payload to a smart contract call submitted via a wallet like Metamask. The data would also be exposed on EthersScan and other block explorers.

That being said looking into it a bit I do not believe there is a reliable way to differentiate between a private and public key protobuf on the ledger and even then as Greg noted it would already be too late exposure wise.

Even if the SDK were able to determine this one could also argue it doesn't account for developers who may decide to utilize other grpc clients to interact with a consensus node.

I'm not sure there's a technical action that can be taken here. 
Non-technically we can affirm in documentation that private keys should never be supplied as function parameters on Hedera.

I'm open minded to other solutions.

**Comment by Kalina-Todorova** [comment]
Hi @ed-marquez as @Nana-EC has mentioned there isn't a technical action to be taken here, so I'm closing the issue.
